packages/api-fetch/src/index.js

```diff
@@ -109,28 +109,28 @@ const defaultFetchHandler = ( nextOptions ) => {
 		}
 	);
 
-	return (
-		responsePromise
-			// Return early if fetch errors. If fetch error, there is most likely no
-			// network connection. Unfortunately fetch just throws a TypeError and
-			// the message might depend on the browser.
-			.then(
-				( value ) =>
-					Promise.resolve( value )
-						.then( checkStatus )
-						.catch( ( response ) =>
-							parseAndThrowError( response, parse )
-						)
-						.then( ( response ) =>
-							parseResponseAndNormalizeError( response, parse )
-						),
-				() => {
-					throw {
-						code: 'fetch_error',
-						message: __( 'You are probably offline.' ),
-					};
-				}
-			)
+	return responsePromise.then(
+		( value ) =>
+			Promise.resolve( value )
+				.then( checkStatus )
+				.catch( ( response ) => parseAndThrowError( response, parse ) )
+				.then( ( response ) =>
+					parseResponseAndNormalizeError( response, parse )
+				),
+		( err ) => {
+			// Return early if fetch throws a TypeError, there is most likely no network connection.
+			// Unfortunately the message might depend on the browser.
+			if ( err instanceof TypeError ) {
```

Comment:
Rather that adding a check for a specific error type for the 80% use case, would it be easier/clearer to determine whether the error _is_ an `AbortError` (ie: the 20% use case) and then return early if that's the case?

Reply:
Yeah it's a good point. I guess the spec isn't really clear of which we should check first (or maybe I'm just missing it), but it makes more sense.